packages/lit-element/src/lib/platform-support.ts

```diff
@@ -0,0 +1,388 @@
+        >;
+        scopeCss.push(
+          ...Array.from(styles).map((style) => {
+            style.parentNode?.removeChild(style);
+            return style.textContent!;
+          })
+        );
+        // Apply styles if this is the result rendering into shadowRoot.
+        if (currentResult.__renderIntoShadowRoot) {
+          if (scopeCss.length) {
+            const style = document.createElement('style');
+            style.textContent = scopeCss.join('\n');
+            // Place aggregate style into template. Note, ShadyCSS will remove
+            // it if polyfilled ShadowDOM is used.
+            template.content.insertBefore(style, template.content.firstChild);
+          }
+          ensureStyles(currentScope, template);
+        }
+      }
+      return template;
+    }
+  }
+
+  Object.assign(NodePart.prototype, {
+    /**
+     * Patch to apply gathered css via ShadyCSS. This is done only once per scope.
+     */
+    __baseSetValue: NodePart.prototype._setValue,
+    _setValue(this: NodePartStandIn, value: unknown) {
+      const container = this._startNode.parentNode!;
+      const renderingIntoShadowRoot = container instanceof ShadowRoot;
+      if (!renderingIntoShadowRoot) {
+        this.__baseSetValue(value);
+      } else {
+        const previousScope = currentScope;
+        currentScope = this.options.scope ?? '';
+
+        const renderingTemplateResult = !!(value as ShadyTemplateResult)
+          ?._$litType$;
+        // If result is a TemplateResult decorate it so that when the template
+        // is created, styling can be shimmed.
+        if (renderingTemplateResult) {
+          (value as ShadyTemplateResult).__renderIntoShadowRoot = renderingIntoShadowRoot;
+        }
+
+        // Note, in first scope render, render into a fragment and move to container
+        // to preserve outer => inner ordering important for @apply production
+        // before consumption.
+        // *** This must be done here in order to support rendering
+        // non-templateResults into shadowRoots.
+        let renderContainer: DocumentFragment | undefined = undefined;
+        const startNode = this._startNode;
+        const endNode = this._endNode;
+        if (renderingIntoShadowRoot && !isScopeStyled(currentScope)) {
+          renderContainer = document.createDocumentFragment();
+          renderContainer.appendChild(document.createComment(''));
+          // Temporarily swizzle tracked nodes so the `__insert` is correct.
+          this._startNode = this._endNode = renderContainer.firstChild!;
+        }
+        this.__baseSetValue(value);
+
+        // Ensure styles are scoped if a TemplateResult is *not* being rendered.
+        // Note, if a TemplateResult *is* being rendered, scoping is done
+        // in `Template._createElement`, before Template parts are created.
+        if (renderingIntoShadowRoot && !renderingTemplateResult) {
+          ensureStyles(currentScope, document.createElement('template'));
```

Comment:
This is so `prepareAdoptedCssText` styles get applied? Comment?

Reply:
Added.